kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/RowTracking.java

```diff
@@ -126,7 +126,10 @@ public void close() throws IOException {
       public CloseableIterator<Row> iterator() {
         // The row ID high watermark from the snapshot of the table that this transaction is reading
         // at. Any baseRowIds higher than this watermark are assigned by this transaction.
-        final long prevRowIdHighWatermark = readRowIdHighWaterMark(txnReadSnapshot);
+        final long prevRowIdHighWatermark =
+            txnReadSnapshot
+                .map(RowTracking::readRowIdHighWaterMark)
```

Comment:
Could we update `readRowIdHighWaterMark` to take in an `Optional<Snapshot>` instead? And then avoid the duplicate logic in these two invoker methods?